lib/services/northBound/deviceProvisioningServer.js

```diff
@@ -316,7 +316,7 @@ function handleRemoveDevice(req, res, next) {
  * This middleware handles updates in the provisioning devices. The only attribute
  */
 function handleUpdateDevice(req, res, next) {
-    if (req.body.deviceId) {
+    if (req.body.device_id) {
```

Comment:
Coding style is to use camelCase not snake_case. Except if there is some important reason (which one?), please use deviceId

Reply:
As per the https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/templates/updateDevice.json#L7 , the property used for device id is device_id. So, I have changed it to device_id.

Reply:
Maybe `device_id` is imposed by API translation  and should be still kept
https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/northBound/deviceProvisioningServer.js#L197

Reply:
> Maybe `deviceId` is imposed by API translation and should be still kept
> https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/northBound/deviceProvisioningServer.js#L197

Hi @AlvaroVega ,
As per my understanding, here the device_id is read from request's body without imposing **toProvisioningAPIFormat(device)** then device_id should be same as per the template https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/templates/updateDevice.json#L7 . Please correct me if my understanding is wrong.

Reply:
I seems that the translate funciton is used only in `res.status(200).json(toProvisioningAPIFormat(device));`, i.e. at response time.

Thus probably you are right @Madhu-NEC and changing from `req.body.deviceId` to `req.body.device_id` here is a good fix.

Reply:
NTC